src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/SafeHandleMarshallingInfoProvider.cs

```diff
@@ -47,18 +51,50 @@ public bool CanProvideMarshallingInfoForType(ITypeSymbol type)
 
         public MarshallingInfo GetMarshallingInfo(ITypeSymbol type, int indirectionDepth, UseSiteAttributeProvider useSiteAttributes, GetMarshallingInfoCallback marshallingInfoCallback)
         {
+            bool hasDefaultConstructor = false;
             bool hasAccessibleDefaultConstructor = false;
             if (type is INamedTypeSymbol named && !named.IsAbstract && named.InstanceConstructors.Length > 0)
             {
                 foreach (IMethodSymbol ctor in named.InstanceConstructors)
                 {
                     if (ctor.Parameters.Length == 0)
                     {
+                        hasDefaultConstructor = true;
                         hasAccessibleDefaultConstructor = _compilation.IsSymbolAccessibleWithin(ctor, _containingScope);
                         break;
                     }
                 }
             }
+
+            if (_safeHandleMarshallerType is not null)
```

Comment:
Let's flip this predicate and using the current return statement here.
```suggestion
            if (_safeHandleMarshallerType is null)
            {
                return new SafeHandleMarshallingInfo(hasAccessibleDefaultConstructor, type.IsAbstract);
            }
```